### PR TITLE
Implement the new count field in the mechanism_spec.

### PR DESCRIPTION
**Pull request by zachferr**
## Description
Implementing count. I haven't added tests yet I want to make sure the implementation looks correct first

## Affected Dependencies
N/A

## How has this been tested?
- No tests yet

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: pipeline_dp/budget_accounting.py [hunk 1]
```diff
@@ -172,19 +174,21 @@ def compute_budgets(self):
 
         total_weight_eps = total_weight_delta = 0
         for mechanism in self._mechanisms:
-            total_weight_eps += mechanism.weight
-            if mechanism.mechanism_spec.use_delta():
-                total_weight_delta += mechanism.weight
+            for i in range(mechanism.mechanism_spec.count):
+                total_weight_eps += mechanism.weight
```
Comment (dvadym) [inline pipeline_dp/budget_accounting.py:178]: Nit:
 total_weight_eps += mechanism.weight*count

and the same with delta

and no need to have for

File: pipeline_dp/budget_accounting.py [hunk 2]
```diff
@@ -172,19 +174,21 @@ def compute_budgets(self):
...
         for mechanism in self._mechanisms:
             eps = delta = 0
-            if total_weight_eps:
-                numerator = self._total_epsilon * mechanism.weight
-                eps = numerator / total_weight_eps
-            if mechanism.mechanism_spec.use_delta():
-                if total_weight_delta:
-                    numerator = self._total_delta * mechanism.weight
-                    delta = numerator / total_weight_delta
+            for i in range(mechanism.mechanism_spec.count):
```
Comment (dvadym) [inline pipeline_dp/budget_accounting.py:184]: No need in this loop. 

**Explanation:** `eps/delta` in `MechanismSpec` is per one application of the mechanism, i.e. no need in count

**Review by dvadym — COMMENTED** [review]
Thanks, it looks good. I left comments. Please add tests

File: tests/budget_accounting_test.py [hunk 3]
```diff
@@ -55,6 +55,22 @@ def test_compute_budgets(self):
...
+                                                  total_delta=1e-6)
+        budget1 = budget_accountant.request_budget(
+            mechanism_type=MechanismType.LAPLACE, weight=4)
+        budget2 = budget_accountant.request_budget(
+            mechanism_type=MechanismType.GAUSSIAN, weight=3, count=2)
+        budget_accountant.compute_budgets()
+
+        self.assertEqual(budget1.eps, 0.4)
+        self.assertEqual(budget1.delta,
+                         0)  # Delta should be 0 if mechanism is Gaussian.
```
[the remark this answers is not in the document]
Comment (zachferr) [reply tests/budget_accounting_test.py]: Fixed in both places.

**Review by dvadym — APPROVED** [review]
Thanks for implementation this!

**Pull request merged**